pkg/report/datafeeds/processor/processor.go

```diff
@@ -0,0 +1,58 @@
+package processor
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder"
+	"github.com/smartcontractkit/chainlink-evm/pkg/monitoring/pb/data-feeds/on-chain/registry"
+	wt "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/monitoring/pb/platform"
+)
+
+// EVM Data-Feeds specific processor decodes writes as 'data-feeds.registry.FeedUpdated' messages + metrics
+type dataFeedsProcessor struct {
+	emitter beholder.ProtoEmitter
+	metrics *registry.Metrics
+	ccip    bool
```

Comment:
imo this just leaks CCIP details here in this type, and makes it handle a special case

vs keeping the type general, and injecting your special case logic with a general type (e.g., inject a specific report decoder for CCIP, vs another one for X)

Reply:
Refactored the Processor logic quite a bit in [this commit](https://github.com/smartcontractkit/chainlink-evm/pull/24/commits/ba6e1bf5dc3be833a611507c5e1157b6ef466cb9) to be injectible and cleaner as you said

Reply:
here you go, hash changed bc of amend sorry about that http://github.com/smartcontractkit/chainlink-evm/pull/24/commits/744075cf446f9482dd6218e418d0b865284c1a02